pkg/pillar/devicenetwork/dns.go

```diff
@@ -39,3 +58,115 @@ func ResolvConfToIfname(resolvConf string) string {
 	}
 	return ""
 }
+
+// ResolveWithSrcIP resolves a domain with a given dns server and source Ip
+func ResolveWithSrcIP(domain string, dnsServerIP net.IP, srcIP net.IP) ([]DNSResponse, error) {
+	var response []DNSResponse
+	sourceUDPAddr := net.UDPAddr{IP: srcIP}
+	dialer := net.Dialer{LocalAddr: &sourceUDPAddr}
+	dnsClient := dns.Client{Dialer: &dialer}
+	msg := dns.Msg{}
+	if domain[len(domain)-1] != '.' {
+		domain = domain + "."
+	}
+	msg.SetQuestion(domain, dns.TypeA)
+	dnsClient.Timeout = time.Duration(dnsTimeout)
+	reply, _, err := dnsClient.Exchange(&msg, net.JoinHostPort(dnsServerIP.String(), "53"))
+	if err != nil {
+		return response, fmt.Errorf("dns exchange failed: %v", err)
+	}
+	for _, answer := range reply.Answer {
+		if aRecord, ok := answer.(*dns.A); ok {
+			response = append(response, DNSResponse{
+				IP:  aRecord.A,
+				TTL: aRecord.Header().Ttl,
+			})
+		}
+	}
+
+	return response, nil
+}
+
+// ResolveWithPortsLambda resolves a domain by using source IPs and dns servers from DeviceNetworkStatus as a resolver func ResolveWithSrcIP can be used
```

Comment:
I would suggest splitting the comment into two sentences for better readability (the seconds starting with `As a...`)